rules/any-pass-simplification.js

```diff
@@ -1,24 +1,24 @@
 'use strict';
+const R = require('ramda');
+const isCalling = require('../ast-helper').isCalling;
 
 const create = context => ({
     CallExpression(node) {
-        if (node.callee.type === 'Identifier'
-            && node.callee.name === 'anyPass'
-            && node.arguments.length > 0
-            && node.arguments[0].type === 'ArrayExpression') {
-            const elements = node.arguments[0].elements;
-            const negated = elements
-                .filter(element =>
-                    element.type === 'CallExpression'
-                    && element.callee.type === 'Identifier'
-                    && element.callee.name === 'complement'
-                );
-            if (elements.length === negated.length) {
-                context.report({
-                    node,
-                    message: '`anyPass` containing only complement functions should be a complement of `allPass`'
-                });
-            }
+        const match = isCalling({
+            name: 'anyPass',
+            arguments: R.both(
+                R.propSatisfies(R.lt(0), 'length'),
+                R.where({
+                    0: R.propSatisfies(R.all(isCalling({ name: 'complement' })), 'elements')
```

Comment:
What happens when the first argument is not an `ArrayExpression`, e.g. `anyPass(myPredicates, myData)`? Looks like the rule would crash in this case.